### ISSUE TITLE
Updating I18N datas

### ISSUE THREAD

**Issue by Pierstoval** [thelia/thelia#1518]
Hi guys, I'm totally new to Thelia, and found many typos in the different texts.
 
Where do I have to update this all to make a PR? Because I saw that many texts are present in both the `setup/insert.sql` file and in the `setup/I18n/*.php` files. Do I have to update both?

**Comment by lunika** [comment]
Hi,

both are independant so you can modify both.

We have a how to contribute : http://doc.thelia.net/en/documentation/contribute.html you can't modify directly a sql file, you have to generate them from a tpl file.

Thanks

**Comment by Pierstoval** [comment]
Hmm. I ran the method and whenever there is a single quote `'` , the "sql generator" changes the string to `null`, which is kind of weird...

**Comment by lunika** [comment]
is this single quote well escaped in the i18n file ? Do you edit the translation file from the admin panel ?

**Comment by Pierstoval** [comment]
Nope, I used the `fr_FR.php` file directly. The quote is escaped the same way other quotes are escaped in the file. Does the `generate:sql` only retrieves database translations?

**Comment by lunika** [comment]
No the command reads the ```fr_FR.php``` file. Can you try to modify the file using the admin panel ? Go to configuration -> translation and select "thelia installation" in the select box.

**Comment by Pierstoval** [comment]
When I use the admin panel and try to update a translation, if there's a quote in it, there's the same problem.
 
Let's take a clear example: 
I want to update this line (it's just an example, there are many lines I want to update in fact):

In the `setup/I18n/fr_FR.php` file:
```diff
-'Login page - after javascript include' => 'Page de connexion - après l\'inclusion des javascript',
+'Login page - after javascript include' => 'Page de connexion - après l\'inclusion du JavaScript'
```

It's after a fresh install so nothing should have been modified in the database since. And in the backoffice I have this:
![capture d ecran 2015-06-22 a 17 15 29](https://cloud.githubusercontent.com/assets/3369266/8285232/517a0190-1902-11e5-8e15-bd5b496316e0.png)

I guess it means that the translation should be OK.

When typing the new translation (view above) in the any redefinition (specific, global, resource), a `local/I18n/fr_FR.php` file is created, and an error occurs because the quote is not well escaped, see:
![capture d ecran 2015-06-22 a 16 57 11](https://cloud.githubusercontent.com/assets/3369266/8285596/9ac8b790-1904-11e5-957e-fd7619acc641.png)

However, for totally unknown reasons, but maybe after 1 change in the database, especially in the `hook_i18n` in this case, I think the system renders the file with a more accurate method because there is no more error. 
 
But if I translate with the backoffice first, the `generate:sql` command works... 
 
Odd?

**Comment by Pierstoval** [comment]
By the way I still made PR #1519 with all manually needed fixes

**Comment by lunika** [comment]
You have to modify the translation in developer mode. The user mode overload the main translation.


![translation](https://cloud.githubusercontent.com/assets/767834/8302742/24b382de-1999-11e5-82c6-4e68a41fe7dd.png)

**Comment by Pierstoval** [comment]
I also tried in developer mode, but then when the text is outputted as `null`, it is totally removed from the file :confused:

**Comment by lunika** [comment]
Which Operating Sytem do you use ? Php version ?

Thanks

**Comment by Pierstoval** [comment]
PHP 5.6.10, Mac OS X Yosemite 10.10.3

**Comment by lunika** [comment]
I test on a mac, the command doesn't work. Thanks

**Comment by lunika** [comment]
or PHP 5.6

**Comment by Pierstoval** [comment]
Maybe it has to be said that I'm using `homebrew`'s php, mysql and apache packages

**Comment by lunika** [comment]
I merged your PR, this is why this issue was closed...

**Comment by Pierstoval** [comment]
Np, it's because it was written "fix #..." in the PR description ;)

**Comment by lunika** [comment]
yes

**Comment by bibich** [comment]
I tested on Mac and have no problems with the command. 
In your previous comment your code was :

```diff
-'Login page - after javascript include' => 'Page de connexion - après l\'inclusion des javascript',
+'Login page - after javascript include' => 'Page de connexion - après l\'inclusion du JavaScript'
```

There is no comma at the end of the new line, that causes the error. 
So I thing that this is not a bug.

It's pretty old, but @Pierstoval could you confirm me ?

**Comment by Pierstoval** [comment]
Actually I have no problem anymore, I think the issue should've been closed after merging #1519 actually

**Comment by bibich** [comment]
ok. thanks.
